### PR TITLE
Enable kubernetes discovery client on bootstrap stage.

### PR DESCRIPTION
**Pull request by wangzw**
This patch enable kubernetes discovery client on bootstrap stage. So that it can works with `spring-cloud-config`

### CODE REVIEW

**Comment by wangzw** [comment]
@spencergibb Added test case.

**Comment by geoand** [comment]
@wangzw Can you please rebase the PR onto master so we can get proper feedback from CI?

**Comment by wangzw** [comment]
Rebased with latest master.

**Comment by geoand** [comment]
LGTM, @spencergibb ?

File: spring-cloud-kubernetes-discovery/src/test/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryAutoConfigurationTests.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import org.springframework.cloud.test.ModifiedClassPathRunner;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * @author Zhanwei Wang
+ */
+@RunWith(ModifiedClassPathRunner.class)
+@ClassPathExclusions({ "spring-retry-*.jar", "spring-boot-starter-aop-*.jar" })
```
Comment (salaboy) [inline spring-cloud-kubernetes-discovery/src/test/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryAutoConfigurationTests.java]: @wangzw @geoand why do you need to exclude these jars here?

Comment (wangzw) [reply spring-cloud-kubernetes-discovery/src/test/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryAutoConfigurationTests.java]: Test case is copied from the link given by @spencergibb

Comment (salaboy) [reply spring-cloud-kubernetes-discovery/src/test/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryAutoConfigurationTests.java]: :+1.. those exclusions should be removed.

**Comment by wangzw** [comment]
Modified as required, please review.

**Pull request merged**